tests/soroban_testcases/storage.rs

```diff
@@ -39,3 +39,89 @@ fn counter() {
     let expected: Val = 10_u64.into_val(&src.env);
     assert!(expected.shallow_eq(&res));
 }
+
+#[test]
+fn different_storage_types() {
+    let src = build_solidity(
+        r#"contract sesa {
```

Comment:
I'd suggest renaming the storage variables `sesa`, `sesa1`, `sesa2`, and `sesa3` to something more descriptive, unless they need to stay like this for some reason

For example, `temporaryCount`, `instanceCount`, and `persistentCount` may help clarify what the variables are intended to test and demonstrate.

I figured i'd mention this because i was trying to figure out what `sesa3` was used for,